### ISSUE TITLE
zed: Failed to open dir "/etc/zfs/zed.d": No such file or directory

### ISSUE THREAD

**Issue by stefanheijnen** [zfsonlinux/pkg-zfs#137]
Yesterday I heard about ZED so I immediately wanted to enable it on my server, but for some reason the dir  "/etc/zfs/zed.d" doesn't exist.

```
beheer@thuis:~$ sudo zed
zed: Failed to open dir "/etc/zfs/zed.d": No such file or directory
beheer@thuis:~$ sudo mkdir /etc/zfs/zed.d/
beheer@thuis:~$ sudo zed
beheer@thuis:~$ ls /etc/zfs/zed.d/
beheer@thuis:~$
```

The only other reference to this issue that I've found is in an IRC log: http://www.mattzone.com/zfsonlinux/zfsonlinux_20141101.html

Linux thuis 3.13.0-34-generic #60-Ubuntu SMP Wed Aug 13 15:45:27 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux

Ubuntu 14.04.1 LTS

ZFS is installed from the official PPA.

```
ii  dkms                                  2.2.0.3-1.1ubuntu5+zfs9~trusty      all          Dynamic Kernel Module Support Framework
rc  libzfs1                               0.6.2-1~precise                     amd64        Native ZFS filesystem library for Linux
ii  libzfs2                               0.6.3-3~trusty                      amd64        Native ZFS filesystem library for Linux
ii  mountall                              2.53-zfs1                           amd64        filesystem mounting tool
ii  ubuntu-zfs                            8~trusty                            amd64        Native ZFS filesystem metapackage for Ubuntu.
ii  zfs-auto-snapshot                     1.1.0-0ubuntu1~trusty               all          ZFS Automatic Snapshot Service
ii  zfs-dkms                              0.6.3-3~trusty                      amd64        Native ZFS filesystem kernel modules for Linux
ii  zfs-doc                               0.6.3-3~trusty                      amd64        Native ZFS filesystem documentation and examples.
ii  zfsutils                              0.6.3-3~trusty                      amd64        Native ZFS management utilities for Linux
ii  spl                                   0.6.3-1~trusty                      amd64        Solaris Porting Layer utilities for Linux
ii  spl-dkms                              0.6.3-1~trusty                      all          Solaris Porting Layer kernel modules for Linux
```

**Comment by dajhorn** [comment]
@stefanheijnen, the PPA will get init scripts for `zed` with the ZoL 0.6.4 release.  Until then, start it through something like `/etc/rc.local` or wedge one of the scripts from the upstream repository.

**Comment by Shingedo** [comment]
I am having the same issue on Ubuntu 14.04 LTS Trusty with version 0.6.4.1 from the PPA

```
ii  dkms                                2.2.0.3-1.1ubuntu5.14.04.1+zfs10~trusty all          Dynamic Kernel Module Support Framework
ii  libzfs2                             0.6.4.1-1~trusty                        amd64        Native OpenZFS filesystem library for Linux
ii  mountall                            2.53-zfs1                               amd64        filesystem mounting tool
ii  ubuntu-zfs                          8~trusty                                amd64        Native ZFS filesystem metapackage for Ubuntu.
ii  zfs-dkms                            0.6.4.1-1~trusty                        amd64        Native OpenZFS filesystem kernel modules for Linux
ii  zfs-doc                             0.6.4.1-1~trusty                        amd64        Native OpenZFS filesystem documentation and examples.
ii  zfsutils                            0.6.4.1-1~trusty                        amd64        Native OpenZFS management utilities for Linux
```

As far as I know the latest 14.04 does not use systemd and also @dajhorn mentioned that this should be fixed in ZoL 0.6.4, but it seems it wasn't.

**Comment by dajhorn** [comment]
> As far as I know the latest 14.04 does not use systemd and also @dajhorn mentioned that this should be fixed in ZoL 0.6.4, but it seems it wasn't.

@EloProf, please install the new `zfs-zed` package, which isn't installed by default because it recommends an MTA.  Also note that the old `/sbin/zed` isn't removed by default.

**Comment by FransUrbo** [comment]
Considering this fixed. Closing.
